packages/cmf/src/component.js

```diff
@@ -0,0 +1,70 @@
+import invariant from 'invariant';
+import action from './action';
+import expression from './expression';
+import registry from './registry';
+import CONST from './constant';
+
+/**
+ * All stuff related to the routing in CMF
+ * @module react-cmf/lib/component
+ */
+
+/**
+ * return a component from the registry
+ * @param  {object} context
+ * @param  {string} id
+ * @return {function} the react component
+ */
+function get(contextOrId, idOrUndefined) {
```

Comment:
This makes the following code had to understand.
As I understand, you introduced anew way to use the get, providing only the id : `get(id)` and the old way to use it is still ok `get(context, id)`
A better idea would be to change the function signature : `get(id, [context])`
This means a breaking change, but if the apps are not using this low level function, the refacto is easy (only in route.js)

@jmfrancois @talend-sromain can you evaluate the difficulty to switch the old calls `route.getComponentFromRegistry(context, id)` to `route.getComponentFromRegistry(id, context)` ?

Reply:
Agree with @jsomsanith, api can and should be simplified

Reply:
ok I will do it !